src/Cache/BlockEsiCache.php

```diff
@@ -53,19 +56,31 @@ class BlockEsiCache extends VarnishCache
      */
     protected $recorder;
 
+    /**
+     * @var ControllerResolverInterface
+     */
+    protected $resolver;
+
+    /**
+     * @var ArgumentResolverInterface
+     */
+    private $argumentResolver;
+
     /**
      * @param string                       $token            A token
      * @param array                        $servers          An array of servers
      * @param RouterInterface              $router           A router instance
      * @param string                       $purgeInstruction The purge instruction (purge in Varnish 2, ban in Varnish 3)
+     * @param ControllerResolverInterface  $resolver         Controller Resolver
+     * @param ArgumentResolverInterface    $argumentResolver Argument Resolver
```

Comment:
I think both of these can be removed (and maybe more, but not in this PR)

Reply:
In this new version, VarnishCache class need

`ControllerResolverInterface $resolver, ArgumentResolverInterface $argumentResolver` 

And it's the same for BlockSsiCache, his extend of SsiCache class need also